DataManagementSystem/DB/FileCatalogComponents/FileMetadata.py

```diff
@@ -330,127 +330,98 @@ def __transformMetaParameterToData( self, metaname ):
-        selectString = ''
-      else:
-        selectString = "%sValue='%s' " % ( table, value )
-
-    return S_OK( selectString )
-
-  def __findFilesForMetaValue( self, meta, value, dirList ):
-    """ Find files in the given list of directories corresponding to the given
-        selection criteria
-    """
-
-    result = self.__createMetaSelection( meta, value, "M." )
-    if not result['OK']:
-      return result
-    selectString = result['Value']
-
-    dirString = ','.join( [ str( x ) for x in dirList] )
-
-    req = " SELECT F.FileID, F.DirID FROM FC_FileMeta_%s AS M, FC_Files AS F" % meta
-    if dirString:
-      req += " WHERE F.DirID in (%s)" % dirString
-    if selectString:
-      if dirString:
-        req += " AND %s AND F.FileID=M.FileID" % selectString
-      else:
-        req += " WHERE %s AND F.FileID=M.FileID" % selectString
-
-
-    result = self.db._query( req )
-    if not result['OK']:
-      return result
-    if not result['Value']:
-      return S_OK( [] )
-
-    fileList = []
-    for row in result['Value']:
-      fileID = row[0]
-      fileList.append( fileID )
-
-    return S_OK( fileList )
-
-  def __findFilesForSE( self, se, dirList ):
-    """ Find files in the given list of directories having replicas in the given se(s)
-    """
-    seList = se
-    if type( se ) in types.StringTypes:
-      seList = [se]
-    seIDs = []
-    for se in seList:
-      result = self.db.seManager.getSEID( se )
-      if not result['OK']:
-        return result
-      seIDs.append( result['Value'] )
-    seString = intListToString( seIDs )
-    dirString = intListToString( dirList )
+            queryList.append( ( '!=', escapedOperand ) )
 
-    req = "SELECT F.FileID FROM FC_Files as F, FC_Replicas as R WHERE F.DirID IN (%s)" % dirString
-    req += " AND R.SEID IN (%s) AND F.FileID=R.FileID" % seString
-    result = self.db._query( req )
-    if not result['OK']:
-      return result
-    if not result['Value']:
-      return S_OK( [] )
-
-    fileList = []
-    for row in result['Value']:
-      fileID = row[0]
-      fileList.append( fileID )
-
-    return S_OK( fileList )
+    return S_OK( queryList )
 
-
-  def __findFilesForStandardMetaValue( self, meta, value, dirList ):
-    """ Find files in the given list of directories corresponding to the given
-        selection criteria using standard file metadata
-    """
-    return S_OK( [] )
-
-  def __buildSEQuery( self, storageElement ):
+  def __buildSEQuery( self, storageElements ):
     """  Return a tuple with table and condition to locate files in a given SE
     """
-    if not storageElement:
+    if not storageElements:
       return S_OK( [] )
-    result = self.db.seManager.getSEID( storageElement )
-    if not result['OK']:
-      return result
-    seID = result['Value']
+
+    seIDList = []
+    for se in storageElements:
+      seID = self.db.seNames.get( se, -1 )
+      if seID == -1:
+        return S_ERROR( 'Unknown SE %s' % se )
```

Comment:
If the SE is missing it will be most likely a typo in the query. I do not think that this should be silently skipped. Propagating a warning to an interactively working user for the otherwise successful request is certainly tricky